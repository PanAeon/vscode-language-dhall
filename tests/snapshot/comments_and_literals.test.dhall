-- This is a line comment
next line should be not comment

-- First comment line
-- Second comment line

in {--} between

{-
multiline
-}
comment should be closed
 
-- Comments can be nested!
{- foo {- bar -} baz -} yes!

--! Better comments should work with my comments

{- TODO:  but no luck, alas -}


'''''''''''''''' -- should interpret as 
                 --  four escape sequences

-- rezerved words:

; A simple label cannot be one of the following reserved names:
;
; * Bool
; * Optional
; * None
; * Natural
; * Integer
; * Double
; * Text
; * List
; * True
; * False
; * NaN
; * Infinity
; * Type
; * Kind
; * Sort
; * Natural/fold
; * Natural/build
; * Natural/isZero
; * Natural/even
; * Natural/odd
; * Natural/toInteger
; * Natural/show
; * Integer/toDouble
; * Integer/show
; * Double/show
; * List/build
; * List/fold
; * List/length
; * List/head
; * List/last
; * List/indexed
; * List/reverse
; * Optional/fold
; * Optional/build
; * if
; * then
; * else
; * let
; * in
; * as
; * using
; * merge
; * constructors
; * Some

-- words
_this_is_single_word
List/indexed -- not sure if this should break on '/' or not

`this.is.a12_-.quot$ed.word`



-- should autoclose:
''well, that's something''
""
[]
<>
()
-- but not in strings
"''"

-- or comm''ents
-12345
+12345
 12345
440394 

-- TODO: Infinity, -Infinity are numbers
Infinity
-Infinity

abcd




