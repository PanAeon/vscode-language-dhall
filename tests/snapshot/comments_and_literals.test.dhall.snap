>-- This is a line comment
#^^ source.dhall comment.line.double-dash.dhall
#  ^^^^^^^^^^^^^^^^^^^^^^^ source.dhall comment.line.double-dash.dhall
>next line should be not comment
#^^^^ source.dhall meta.label.dhall
#    ^ source.dhall
#     ^^^^ source.dhall meta.label.dhall
#         ^ source.dhall
#          ^^^^^^ source.dhall meta.label.dhall
#                ^ source.dhall
#                 ^^ source.dhall meta.label.dhall
#                   ^ source.dhall
#                    ^^^ source.dhall meta.label.dhall
#                       ^ source.dhall
#                        ^^^^^^^ source.dhall meta.label.dhall
>
>-- First comment line
#^^ source.dhall comment.line.double-dash.dhall
#  ^^^^^^^^^^^^^^^^^^^ source.dhall comment.line.double-dash.dhall
>-- Second comment line
#^^ source.dhall comment.line.double-dash.dhall
#  ^^^^^^^^^^^^^^^^^^^^ source.dhall comment.line.double-dash.dhall
>
>in {--} between
#^^ source.dhall keyword.control.dhall
#  ^ source.dhall
#   ^^ source.dhall comment.block.dhall
#     ^^ source.dhall comment.block.dhall
#       ^ source.dhall
#        ^^^^^^^ source.dhall meta.label.dhall
>
>{-
#^^ source.dhall comment.block.dhall
>multiline
#^^^^^^^^^^ source.dhall comment.block.dhall
>-}
#^^ source.dhall comment.block.dhall
>comment should be closed
#^^^^^^^ source.dhall meta.label.dhall
#       ^ source.dhall
#        ^^^^^^ source.dhall meta.label.dhall
#              ^ source.dhall
#               ^^ source.dhall meta.label.dhall
#                 ^ source.dhall
#                  ^^^^^^ source.dhall meta.label.dhall
> 
>-- Comments can be nested!
#^^ source.dhall comment.line.double-dash.dhall
#  ^^^^^^^^^^^^^^^^^^^^^^^^ source.dhall comment.line.double-dash.dhall
>{- foo {- bar -} baz -} yes!
#^^ source.dhall comment.block.dhall
#  ^^^^^ source.dhall comment.block.dhall
#       ^^ source.dhall comment.block.dhall comment.block.dhall
#         ^^^^^ source.dhall comment.block.dhall comment.block.dhall
#              ^^ source.dhall comment.block.dhall comment.block.dhall
#                ^^^^^ source.dhall comment.block.dhall
#                     ^^ source.dhall comment.block.dhall
#                       ^ source.dhall
#                        ^^^ source.dhall meta.label.dhall
#                           ^^ source.dhall
>
>--! Better comments should work with my comments
#^^ source.dhall comment.line.double-dash.dhall
#  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.dhall comment.line.double-dash.dhall
>
>{- TODO:  but no luck, alas -}
#^^ source.dhall comment.block.dhall
#  ^^^^^^^^^^^^^^^^^^^^^^^^^^ source.dhall comment.block.dhall
#                            ^^ source.dhall comment.block.dhall
>
>
>'''''''''''''''' -- should interpret as 
#^^^^^^^^^^^^^^^^^ source.dhall
#                 ^^ source.dhall comment.line.double-dash.dhall
#                   ^^^^^^^^^^^^^^^^^^^^^ source.dhall comment.line.double-dash.dhall
>                 --  four escape sequences
#^^^^^^^^^^^^^^^^^ source.dhall
#                 ^^ source.dhall comment.line.double-dash.dhall
#                   ^^^^^^^^^^^^^^^^^^^^^^^ source.dhall comment.line.double-dash.dhall
>
>-- rezerved words:
#^^ source.dhall comment.line.double-dash.dhall
#  ^^^^^^^^^^^^^^^^ source.dhall comment.line.double-dash.dhall
>
>; A simple label cannot be one of the following reserved names:
#^^ source.dhall
#  ^ source.dhall meta.label.dhall
#   ^ source.dhall
#    ^^^^^^ source.dhall meta.label.dhall
#          ^ source.dhall
#           ^^^^^ source.dhall meta.label.dhall
#                ^ source.dhall
#                 ^^^^^^ source.dhall meta.label.dhall
#                       ^ source.dhall
#                        ^^ source.dhall meta.label.dhall
#                          ^ source.dhall
#                           ^^^ source.dhall meta.label.dhall
#                              ^ source.dhall
#                               ^^ source.dhall meta.label.dhall
#                                 ^ source.dhall
#                                  ^^^ source.dhall meta.label.dhall
#                                     ^ source.dhall
#                                      ^^^^^^^^^ source.dhall meta.label.dhall
#                                               ^ source.dhall
#                                                ^^^^^^^^ source.dhall meta.label.dhall
#                                                        ^ source.dhall
#                                                         ^^^^^ source.dhall meta.label.dhall
#                                                              ^ source.dhall keyword.other.colon.dhall
>;
#^^ source.dhall
>; * Bool
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^ source.dhall support.class.dhall
>; * Optional
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^ source.dhall support.class.dhall
>; * None
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^ source.dhall support.class.dhall
>; * Natural
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^ source.dhall support.class.dhall
>; * Integer
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^ source.dhall support.class.dhall
>; * Double
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^ source.dhall entity.name.function
>; * Text
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^ source.dhall support.class.dhall
>; * List
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^ source.dhall support.class.dhall
>; * True
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^ source.dhall constant.language.dhall
>; * False
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^ source.dhall constant.language.dhall
>; * NaN
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^ source.dhall constant.language.dhall
>; * Infinity
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^ source.dhall constant.language.dhall
>; * Type
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^ source.dhall keyword.other.dhall
>; * Kind
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^ source.dhall keyword.other.dhall
>; * Sort
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^ source.dhall keyword.other.dhall
>; * Natural/fold
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^^ source.dhall entity.name.function
>; * Natural/build
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^^^ source.dhall entity.name.function
>; * Natural/isZero
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^^^^ source.dhall entity.name.function
>; * Natural/even
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^^ source.dhall entity.name.function
>; * Natural/odd
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^ source.dhall entity.name.function
>; * Natural/toInteger
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^^^^^^^ source.dhall entity.name.function
>; * Natural/show
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^^ source.dhall entity.name.function
>; * Integer/toDouble
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^^^^^^ source.dhall entity.name.function
>; * Integer/show
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^^ source.dhall entity.name.function
>; * Double/show
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^ source.dhall entity.name.function
#          ^^^^^ source.dhall string.unquoted.file.dhall meta.path.file.dhall
>; * List/build
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^ source.dhall entity.name.function
>; * List/fold
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^ source.dhall entity.name.function
>; * List/length
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^ source.dhall entity.name.function
>; * List/head
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^ source.dhall entity.name.function
>; * List/last
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^ source.dhall entity.name.function
>; * List/indexed
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^^ source.dhall entity.name.function
>; * List/reverse
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^^ source.dhall entity.name.function
>; * Optional/fold
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^^^ source.dhall entity.name.function
>; * Optional/build
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^^^^ source.dhall entity.name.function
>; * if
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^ source.dhall keyword.control.conditional.dhall
>; * then
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^ source.dhall keyword.control.conditional.dhall
>; * else
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^ source.dhall keyword.control.conditional.dhall
>; * let
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^ source.dhall meta.declaration.expression.let.dhall keyword.other.let.dhall
>; * in
#^^^^ source.dhall meta.declaration.expression.let.dhall
#    ^^ source.dhall keyword.control.dhall
>; * as
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^ source.dhall keyword.control.dhall
>; * using
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^ source.dhall keyword.control.dhall
>; * merge
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^ source.dhall keyword.control.dhall
>; * constructors
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^^^^^^^^^ source.dhall keyword.control.dhall
>; * Some
#^^ source.dhall
#  ^ source.dhall keyword.operator.arithmetic.dhall
#   ^ source.dhall
#    ^^^^ source.dhall constant.language.dhall
>
>-- words
#^^ source.dhall comment.line.double-dash.dhall
#  ^^^^^^ source.dhall comment.line.double-dash.dhall
>_this_is_single_word
#^^^^^^^^^^^^^^^^^^^^ source.dhall meta.label.dhall
>List/indexed -- not sure if this should break on '/' or not
#^^^^^^^^^^^^ source.dhall entity.name.function
#            ^ source.dhall
#             ^^ source.dhall comment.line.double-dash.dhall
#               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.dhall comment.line.double-dash.dhall
>
>`this.is.a12_-.quot$ed.word`
#^ source.dhall
# ^^^^ source.dhall meta.label.dhall
#     ^ source.dhall
#      ^^ source.dhall meta.label.dhall
#        ^ source.dhall
#         ^^^^^ source.dhall meta.label.dhall
#              ^ source.dhall
#               ^^^^ source.dhall meta.label.dhall
#                   ^ source.dhall
#                    ^^ source.dhall meta.label.dhall
#                      ^ source.dhall
#                       ^^^^ source.dhall meta.label.dhall
#                           ^^ source.dhall
>
>
>
>-- should autoclose:
#^^ source.dhall comment.line.double-dash.dhall
#  ^^^^^^^^^^^^^^^^^^ source.dhall comment.line.double-dash.dhall
>''well, that's something''
#^^ source.dhall
#  ^^^^ source.dhall meta.label.dhall
#      ^^ source.dhall
#        ^^^^ source.dhall meta.label.dhall
#            ^ source.dhall
#             ^ source.dhall meta.label.dhall
#              ^ source.dhall
#               ^^^^^^^^^ source.dhall meta.label.dhall
#                        ^^ source.dhall string.quoted.single.dhall
>""
#^^^ source.dhall string.quoted.single.dhall
>[]
#^^^ source.dhall string.quoted.single.dhall
><>
#^^^ source.dhall string.quoted.single.dhall
>()
#^^^ source.dhall string.quoted.single.dhall
>-- but not in strings
#^^^^^^^^^^^^^^^^^^^^^^ source.dhall string.quoted.single.dhall
>"''"
#^ source.dhall string.quoted.single.dhall
# ^^ source.dhall string.quoted.single.dhall
#   ^ source.dhall string.quoted.double.dhall
>
>-- or comm''ents
#^^^^^^^^^^^^^^^^^ source.dhall string.quoted.double.dhall
>-12345
#^^^^^^^ source.dhall string.quoted.double.dhall
>+12345
#^^^^^^^ source.dhall string.quoted.double.dhall
> 12345
#^^^^^^^ source.dhall string.quoted.double.dhall
>440394 
#^^^^^^^^ source.dhall string.quoted.double.dhall
>
>-- TODO: Infinity, -Infinity are numbers
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.dhall string.quoted.double.dhall
>Infinity
#^^^^^^^^^ source.dhall string.quoted.double.dhall
>-Infinity
#^^^^^^^^^^ source.dhall string.quoted.double.dhall
>
>abcd
#^^^^^ source.dhall string.quoted.double.dhall
>
>
>
>
>